From 20b3c7c8b3653940f357729e9f7159e2f32c1021 Mon Sep 17 00:00:00 2001
From: Timo Teras <timo.teras@iki.fi>
Date: Thu, 6 Aug 2009 13:09:38 +0300
Subject: [PATCH 5/7] audit: do not load repositories

they are not needed, and cause errors if signing keys are not
present.
---
 src/audit.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/audit.c b/src/audit.c
index e47f182..67cefac 100644
--- a/src/audit.c
+++ b/src/audit.c
@@ -19,6 +19,7 @@
 #include "apk_database.h"
 
 struct audit_ctx {
+	unsigned int open_flags;
 	int (*audit)(struct apk_database *db);
 };
 
@@ -169,9 +170,11 @@ static int audit_main(void *ctx, int argc, char **argv)
 	if (actx->audit == NULL)
 		return -EINVAL;
 
-	r = apk_db_open(&db, apk_root, APK_OPENF_READ);
+	r = apk_db_open(&db, apk_root,
+		        APK_OPENF_READ | APK_OPENF_NO_SCRIPTS |
+			APK_OPENF_NO_REPOS);
 	if (r != 0) {
-		apk_error("APK database not present");
+		apk_error("Unable to open db: %s", apk_error_str(r));
 		return r;
 	}
 	r = actx->audit(&db);
-- 
1.6.4

